spec/lang/representation.md

```diff
@@ -105,25 +115,45 @@ fn decode_ptr<M: Memory>(bytes: List<AbstractByte<M::Provenance>>) -> Option<Poi
             provenance = None;
         }
     }
-    ret(Pointer { addr, provenance })
+    ret(ThinPointer { addr, provenance })
 }
 
-fn encode_ptr<M: Memory>(ptr: Pointer<M::Provenance>) -> List<AbstractByte<M::Provenance>> {
+fn encode_ptr<M: Memory>(ptr: ThinPointer<M::Provenance>) -> List<AbstractByte<M::Provenance>> {
     let bytes_data = M::T::ENDIANNESS.encode(Unsigned, M::T::PTR_SIZE, ptr.addr).unwrap();
     bytes_data.map(|b| AbstractByte::Init(b, ptr.provenance))
+    
+}
+
+fn decode_ptr_meta<M: Memory>(kind: PointerMetaKind, bytes: List<AbstractByte<M::Provenance>>) -> Option<Option<PointerMeta>> {
+    match kind {
+        PointerMetaKind::None if bytes.is_empty() => ret(None),
+        _ => { throw!(); }
+    }
+}
+
+fn encode_ptr_meta<M: Memory>(meta: Option<PointerMeta>) -> List<AbstractByte<M::Provenance>> {
+    match meta {
+        None => list!(),
+        Some(_) => unreachable!("TODO: others"),
+    }
 }
```

Comment:
I don't understand the point of adding these functions, they are quite confusing.

I think pointer encode/decode should stay largely unchanged in this PR, and simply set the metadata field to `None` and assert that it is `None` in `encode`.

Reply:
Fair, this was an artifact of the future